mlir/lib/Dialect/Mesh/Interfaces/ShardingInterface.cpp

```diff
@@ -168,16 +168,16 @@ LogicalResult mesh::ShardingInterface::verifyShardingInterfaceImpl() {
 
   // check operands and results type
   for (Type type : op->getOperandTypes())
-    if (!llvm::isa<RankedTensorType>(type))
+    if (!llvm::isa<RankedTensorType>(type) && !type.isIntOrIndexOrFloat())
```

Comment:
So unranked types of non-int/float are ok?

Reply:
Does any `UnrankedTensorType` ever return true on `isIntOrIndexOrFLoat()` ? This is trying to fail for everything which is not a `RankedTensor` and not a scalar.

Reply:
Ah, probably not, just their element types.